src/Microsoft.AspNetCore.Razor.Language/RazorProjectEngine.cs

```diff
@@ -3,26 +3,63 @@
 
 using System;
 using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language.Extensions;
```

Comment:
Take a beat to look at what I did here, because I expanded the surface of this class significantly, but it's mostly convenience, clarity, and surfacing things we already have.